ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -685,26 +684,21 @@ private void writeMessagesToLocalStoreAndAdvanceTokens(List<ExchangeMetadataResp
                   exchangeMetadataResponse.missingStoreKeys, remoteReplicaInfo.getReplicaId().getPartitionId(),
                   remoteReplicaInfo.getLocalReplicaId().getMountPath());
 
-              MessageFormatWriteSet writeset = null;
-              if (validateMessageStream) {
-                MessageSievingInputStream validMessageDetectionInputStream =
-                    new MessageSievingInputStream(getResponse.getInputStream(), messageInfoList, storeKeyFactory,
-                        metricRegistry);
-                if (validMessageDetectionInputStream.hasInvalidMessages()) {
-                  replicationMetrics.incrementInvalidMessageError(partitionResponseInfo.getPartition());
-                  logger.error("Out of " + (messageInfoList.size()) + " messages, " + (messageInfoList.size()
-                      - validMessageDetectionInputStream.getValidMessageInfoList().size())
-                      + " invalid messages were found in message stream from " + remoteReplicaInfo.getReplicaId());
-                }
-                messageInfoList = validMessageDetectionInputStream.getValidMessageInfoList();
-                if (messageInfoList.size() == 0) {
-                  logger.error("MessageInfoList is of size 0 as all messages are invalidated ");
-                } else {
-                  writeset = new MessageFormatWriteSet(validMessageDetectionInputStream, messageInfoList, false);
-                  remoteReplicaInfo.getLocalStore().put(writeset);
-                }
+              MessageFormatWriteSet writeset;
+              MessageSievingInputStream validMessageDetectionInputStream =
+                  new MessageSievingInputStream(getResponse.getInputStream(), messageInfoList, transformers,
+                      metricRegistry);
+              if (validMessageDetectionInputStream.hasInvalidMessages()) {
+                replicationMetrics.incrementInvalidMessageError(partitionResponseInfo.getPartition());
+                logger.error("Out of " + (messageInfoList.size()) + " messages, " + (messageInfoList.size()
+                    - validMessageDetectionInputStream.getValidMessageInfoList().size())
+                    + " invalid messages were found in message stream from " + remoteReplicaInfo.getReplicaId());
+              }
+              messageInfoList = validMessageDetectionInputStream.getValidMessageInfoList();
+              if (messageInfoList.size() == 0) {
+                logger.error("MessageInfoList is of size 0 as all messages are invalidated ");
```

Comment:
this may no longer be true - it could also be because they are deprecated?

Reply:
needs addressing

Reply:
Ah - I addressed this in another file which had a similar error. Fixed now.